src/utilities/paramValidation.ts

```diff
@@ -0,0 +1,22 @@
+import { Param, RouteFlat } from '@/types'
+import { getParamValue, findParamValues } from '@/utilities'
+
+export function routeParamsAreValid(url: string, route: RouteFlat): boolean {
+  const params = Object.entries<Param[]>(route.params)
+
+  for (const [key, paramsTuple] of params) {
+    const stringValues = findParamValues(url, route.path, key)
```

Comment:
Are these always strings or could undefined exist in this array?

Reply:
you're right that in the previous configuration these values were `(string | undefined)[]`, but after making these code review suggestions it's actually all strings, `undefined` values are coming back as an empty string, which is a requirement after changing `getParamValue` back to only accepting a string value

Reply:
That makes sense to me 👍